api/src/services/spatial-service.ts

```diff
@@ -0,0 +1,102 @@
+import { IDBConnection } from '../database/db';
+import { IGetSpatialTransformRecord, SpatialRepository } from '../repositories/spatial-repository';
+import { DBService } from './db-service';
+
+export class SpatialService extends DBService {
+  spatialRepository: SpatialRepository;
+
+  constructor(connection: IDBConnection) {
+    super(connection);
+
+    this.spatialRepository = new SpatialRepository(connection);
+  }
+
+  /**
+   * get spatial transform record from name
+   *
+   * @param {string} spatialTransformName
+   * @return {*}  {Promise<IGetSpatialTransformRecord>}
+   * @memberof SpatialService
+   */
+  async getSpatialTransformRecords(): Promise<IGetSpatialTransformRecord[]> {
+    return this.spatialRepository.getSpatialTransformRecords();
+  }
+
+  /**
+   * Insert record of transform id used for submission spatial component record
+   *
+   * @param {number} spatialTransformId
+   * @param {number} submissionSpatialComponentId
+   * @return {*}  {Promise<{ spatial_transform_submission_id: number }>}
+   * @memberof SpatialService
+   */
+  async insertSpatialTransformSubmissionRecord(
+    spatialTransformId: number,
+    submissionSpatialComponentId: number
+  ): Promise<{ spatial_transform_submission_id: number }> {
+    return this.spatialRepository.insertSpatialTransformSubmissionRecord(
+      spatialTransformId,
+      submissionSpatialComponentId
+    );
+  }
+
+  /**
+   * Collect transforms from db, run transformations on submission id, save result to spatial component table
+   *
+   * @param {number} submissionId
+   * @return {*}  {Promise<void>}
+   * @memberof SpatialService
+   */
+  async runSpatialTransforms(submissionId: number): Promise<void> {
+    const spatialTransformRecords = await this.getSpatialTransformRecords();
+
+    const promises1 = spatialTransformRecords.map(async (transformRecord) => {
```

Comment:
Could there be a more apt name for these two promise arrays than `promises1` and `promises2`, like maybe `transformRecordPromises`?